lib/Backend/NativeCodeGenerator.cpp

```diff
@@ -1117,7 +1117,7 @@ NativeCodeGenerator::CodeGen(PageAllocator * pageAllocator, CodeGenWorkItem* wor
 
     if (!CONFIG_FLAG(OOPCFGRegistration))
     {
-        scriptContext->GetThreadContext()->SetValidCallTargetForCFG((PVOID)jitWriteData.codeAddress);
```

Comment:
I think this is only a partial fix. There is a different place to do this if config flag is set. And presumably the unregister call also needs it.